options/options_test.go

```diff
@@ -244,6 +254,25 @@ func TestOptionsInUnlinkedFiles(t *testing.T) {
 	}
 }
 
+func TestOptionsInUnlinkedFileInvalid(t *testing.T) {
```

Comment:
This is a repro test case. It differs from the ones above because it actually is expected to return an error, even in lenient mode (since the error is related to an unresolved option or type error). This would previously trigger the panic. And then, after the panic was fixed, it would return the wrong error without the other changes in this branch.